### PR TITLE
ENG-13675: update documentation for identity_provider field

### PR DESCRIPTION
**Pull request by ricardorey10**
## Description of the change

As title. Part of https://cyralinc.atlassian.net/wiki/spaces/EN/pages/2212921352/Deprecate+IdP+Specification+in+Access+Token+Based+Repo+Authentication

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] Jira issue referenced in commit message and/or PR title

### Testing

> Description of testing

### CODE REVIEW

**Review by juniocezar — COMMENTED** [review]
looks ok to me, but I'll defer to Wilson and Deepak on this one

**Review by yoursnerdly — APPROVED** [review]

**Comment by wcmjunior** [comment]
Please do not merge. I would like to discuss this change internally before we move. Thanks.

**Comment by ricardorey10** [comment]
> Please do not merge. I would like to discuss this change internally before we move. Thanks.

@wcmjunior sure thing. I'll wait for you to let me know what to do. Thanks!

File: cyral/internal/repository/confauth/resource_cyral_repository_conf_auth.go
Comment (ricardorey10) [inline cyral/internal/repository/confauth/resource_cyral_repository_conf_auth.go:220]: @wcmjunior please suggest here how the phrasing should look like :grin:

File: docs/resources/repository_conf_auth.md
Comment (wcmjunior) [inline docs/resources/repository_conf_auth.md:43]: @ricardorey10 this is how it should *look like*, so you will have to adjust your `go` file to be able to generate such output:

```suggestion
- `identity_provider` (String) The semantics of this field changed in control planes `v4.14` and later. See how it should be configured depending on your control plane version:
  - `v4.13` and below:
     - Provide the ID (Alias) of the identity provider integration to allow user authentication using an IdP.
  - `v4.14` and later:
     - If not supplied, then end-user authentication is disabled.
     - If end-user authentication with Cyral Access Token is desired, then set to `ACCESS_TOKEN` or any other non-empty string.
     - If end-user authentication with AWS IAM is desired, then this must be the ID of an AWS IAM integration, and the `auth_type` attribute must be set to `AWS_IAM`.
```

This will generate this output, which is much easier to read: 
image

Use the [Terraform docs preview](https://registry.terraform.io/tools/doc-preview) tool to see how your MD files will look like in the final docs shown in the Terraform registry.

Comment (ricardorey10) [reply docs/resources/repository_conf_auth.md:43]: Thanks for the feedback! We're releasing this in 4.13, so I'll update the versions a bit, but otherwise I'll copy the text pretty much verbatim.

I'll make sure that the output looks correct before merging!

**Review by yoursnerdly — APPROVED** [review]

**Review by VictorGFM — APPROVED** [review]

**Pull request merged**
